components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/factory/KeyManagerHolder.java

```diff
@@ -71,51 +71,52 @@ public static void addKeyManagerConfiguration(String tenantDomain, String name,
             log.error("Key Manager " + name + " already initialized in tenant " + tenantDomain);
         }
         if (keyManagerConfiguration.isEnabled()) {
-            KeyManager keyManager = null;
-            JWTValidator jwtValidator = null;
-            APIManagerConfiguration apiManagerConfiguration =
-                    ServiceReferenceHolder.getInstance().getAPIManagerConfigurationService()
-                            .getAPIManagerConfiguration();
-            String defaultKeyManagerType =
-                    apiManagerConfiguration.getFirstProperty(APIConstants.DEFAULT_KEY_MANAGER_TYPE);
-            KeyManagerConnectorConfiguration keyManagerConnectorConfiguration =
-                    ServiceReferenceHolder.getInstance().getKeyManagerConnectorConfiguration(type);
-            if (keyManagerConnectorConfiguration != null) {
-                if (StringUtils.isNotEmpty(keyManagerConnectorConfiguration.getImplementation())) {
-                    try {
-                        keyManager = (KeyManager) Class
-                                .forName(keyManagerConnectorConfiguration.getImplementation()).newInstance();
-                        keyManager.setTenantDomain(tenantDomain);
-                        if (StringUtils.isNotEmpty(defaultKeyManagerType) && defaultKeyManagerType.equals(type)){
-                            keyManagerConfiguration.addParameter(APIConstants.KEY_MANAGER_USERNAME,
-                                    apiManagerConfiguration.getFirstProperty(APIConstants.API_KEY_VALIDATOR_USERNAME));
-                            keyManagerConfiguration.addParameter(APIConstants.KEY_MANAGER_PASSWORD,
-                                    apiManagerConfiguration.getFirstProperty(APIConstants.API_KEY_VALIDATOR_PASSWORD));
+            if (!KeyManagerConfiguration.TokenType.EXCHANGED.equals(keyManagerConfiguration.getTokenType())) {
```

Comment:
Let's negate if and do a return from beginning

Reply:
Fixed via https://github.com/wso2/carbon-apimgt/pull/10567/commits/e1f31cb54fdf865a9fefa95e4ce9dd8bda815842